src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -115,6 +119,7 @@ func (mod *mgmtModule) handleGetAttachInfo(reqb []byte, pid int32) ([]byte, erro
 	// caching is enabled, there's data in the info cache and the agent can quickly return
 	// a response without the overhead of a mutex.
 	if mod.aiCache.isCached() {
+		mod.monitor.submitRequest(pid, drpc.MethodGetAttachInfo)
```

Comment:
As I was revisiting this, I started wondering if it was a bad idea to overload these drpc method IDs. Started to confuse myself that we were making drpc calls here, though we're not. This is really just signaling the monitor to watch a process. Maybe it would be better to have the calls at this level be something like monitor.Register(pid) and monitor.Unregister(pid). The implementation underneath could stay basically the same, just would be less confusing reading the caller code. And since attach and disconnect are the only actions we care about, it seems like procMon could have its own enum to represent those actions in the request object, instead of relying on drpc.

Reply:
I originally thought of doing this but attach and disconnect wont be the only actions. We're going to have to register the pool handles during connect and then remove the handle from the list on pool disconnect as well. These will map directly to drpc calls so we'll have a second list of IDs that can get out of sync and will either have the same numbers as the MgmtMethod item or we will have to have a translation method between the drpc call and the identifier inside here.

Reply:
I guess I don't totally understand - we're not pulling the drpc method ID from the request, we just know what method we're already in. Won't that be true for the pool connect as well? Or am I missing something?

Reply:
For what it's worth, I do think a bit of abstraction as Kris suggests here would make the code easier to read and reason about. Otherwise it means that an unfamiliar reader in this code will have to go look at the actual implementation to figure out what's going on.

Reply:
Ok so is the idea to:

1) Undo the export of MgmtMethod.
2) Make a new enum for the requests type which isn't exported.
3) Make individual methods for submitting the types of request like RegisterProcess, UnregisterProcess, RegisterPoolHandle, UnregisterPoolHandle?

Reply:
I think just option 3. Behind the scenes you can just keep the current implementation, it's just hiding it behind methods with names that make their purpose immediately clear.

Reply:
Fixed